src/nvim/message.c

```diff
@@ -137,6 +139,81 @@ int msg_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
   return msg_attr_keep((char_u *)s, attr, false);
 }
 
+int msg_echo_attr(const char *s, const int attr) FUNC_ATTR_NONNULL_ARG(1)
+{
+  return msg_echo_attr_keep(s, attr, false);
+}
+
+int
+msg_echo_attr_keep(const char *s, const int attr, int keep)
+  FUNC_ATTR_NONNULL_ALL
+{
+  static int entered = 0;
+  int retval;
+  char * buf = NULL;
+
+  if (!emsg_on_display && message_filtered((char_u *)s)) {
+    return true;
+  }
+
+  if (attr == 0) {
+    set_vim_var_string(VV_STATUSMSG, s, -1);
+  }
+
+  if (entered >= 3) {
+    return true;
+  }
+  entered++;
+
+  if ((char_u *)s == keep_msg) {
+    keep_msg = NULL;
+  }
+
+  msg_start();
+  buf = (char *)msg_strtrunc((char_u *)s, false);
+  if (buf != NULL) {
+    s = buf;
+  }
+
+  const char *spec_char = "\t\n\r";
```

Comment:
This variable still has no reason to exist.

Reply:
Oh I see, I misunderstood your previous comment